application/src/main/java/org/thingsboard/server/service/edge/EdgeKafkaLagReporter.java

```diff
@@ -0,0 +1,163 @@
+                    .filter(id -> id.startsWith("tb_edge_event.notifications."))
+                    .collect(Collectors.toSet());
+
+            log.debug("Found {} consumer groups with prefix 'tb_edge_event.notifications.'", groupIds.size());
+
+            for (String groupId : groupIds) {
+                log.debug("Processing consumer group: {}", groupId);
+
+                ListConsumerGroupOffsetsResult offsetsResult = adminClient.listConsumerGroupOffsets(groupId);
+                Map<TopicPartition, OffsetAndMetadata> groupOffsets = offsetsResult.partitionsToOffsetAndMetadata().get(5, TimeUnit.SECONDS);
+                Map<TopicPartition, Long> endOffsets = consumer.endOffsets(groupOffsets.keySet());
+
+                for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : groupOffsets.entrySet()) {
+                    TopicPartition tp = entry.getKey();
+                    String topic = tp.topic();
+                    long committed = entry.getValue().offset();
+                    Long latest = endOffsets.get(tp);
+                    if (latest == null) {
+                        log.warn("No end offset for topic-partition {}. Skipping.", tp);
+                        continue;
+                    }
+
+                    long lag = latest - committed;
+
+                    extractTenantEdge(topic).ifPresent(tenantEdge -> {
+                        long now = System.currentTimeMillis();
+                        long roundedTs = now - (now % TimeUnit.MINUTES.toMillis(1));
+                        TsKvEntry kvEntry = new BasicTsKvEntry(roundedTs, new LongDataEntry(LAG_KEY, lag));
+
+                        ListenableFuture<TimeseriesSaveResult> future = tsService.save(
+                                TenantId.fromUUID(tenantEdge.getFirst()),
+                                new EdgeId(tenantEdge.getSecond()),
+                                kvEntry
+                        );
+
+                        Futures.addCallback(future, new com.google.common.util.concurrent.FutureCallback<>() {
+                            @Override
+                            public void onSuccess(TimeseriesSaveResult result) {
+                                log.debug("Successfully saved downlinkLag [{}] for edge [{}]", lag, tenantEdge.getSecond());
+                            }
+
+                            @Override
+                            public void onFailure(Throwable t) {
+                                log.warn("Failed to save downlinkLag for edge [{}]: {}", tenantEdge.getSecond(), t.getMessage(), t);
+                            }
+                        }, MoreExecutors.directExecutor());
+                    });
+                }
+            }
+        } catch (Exception e) {
+            log.warn("Error reporting Edge Kafka lag: ", e);
+        }
+    }
+
+    /**
+     * Parse tenantId and edgeId from topic name.
+     * Expected format: tb_edge_event.notifications.<tenantId>.<edgeId>
```

Comment:
please check what is expected format in case TB_QUEUE_PREFIX and TB_QUEUE_EDGE_EVENT_NOTIFICATIONS_TOPIC are set and if they have more . then expected in this code?